lib/foreman/renderer.rb

```diff
@@ -57,6 +59,11 @@ def foreman_url(action = "built")
               :token     => (@host.token.value unless @host.try(:token).nil?), :kind => action
     end
 
+    def foreman_server_fqdn
+      config = URI.parse(Setting[:unattended_url])
+      config.host || request.host
```

Comment:
ok, I can add `foreman_server_url` that will add protocol and port too (no path, no template proxies in this one), would that help?